lib/internal/priority_queue.js

```diff
@@ -73,27 +81,31 @@ module.exports = class PriorityQueue {
     const item = heap[pos];
 
     while (pos > 1) {
-      const parent = heap[pos / 2 | 0];
-      if (compare(parent, item) <= 0)
+      const parent = pos >> 1;
+      const parentItem = heap[parent];
+      if (compare(parentItem, item) <= 0)
         break;
-      heap[pos] = parent;
+      heap[pos] = parentItem;
       if (setPosition !== undefined)
-        setPosition(parent, pos);
-      pos = pos / 2 | 0;
+        setPosition(parentItem, pos);
+      pos = parent;
     }
+
     heap[pos] = item;
     if (setPosition !== undefined)
       setPosition(item, pos);
   }
 
   removeAt(pos) {
+    if (pos > this.#size) return;
```

Comment:
Wouldn't an assert makes more sense here? We probably have a bug if we try to remove something out of range
```suggestion
    assert (pos < this.#size);
```

Reply:
@aduh95 actually I checked again and it was going to be a breaking change this one, because of this:

https://github.com/nodejs/node/blob/25dd206c29f232bd5445ffb96e517b8baa69adc1/lib/internal/priority_queue.js#L89-L101

We'd do the mutation directly but wouldn't enter the if block, now we don't do anything, so I reverted that completely